### DISCUSSION TITLE
[mini.notify]: send notification on save

### DISCUSSION THREAD

**Discussion by pwnwriter** [nvim-mini/mini.nvim#980] [General]
Previously I used to use [`folke/noice.nvim`](https://github.com/folke/noice.nvim) which used to completely remove other messages and redirect them to send notification . It is possible to send notification using `mini.notify` ? 

For example I'd like to remove the following message  and instead send notification? 

<img width="367" alt="Screenshot 2024-06-19 at 15 03 09" src="https://github.com/echasnovski/mini.nvim/assets/90331517/1caac91a-4351-48ea-8483-4b8af355194d">

**Comment by echasnovski** [comment]
I am afraid, this is not possible to the extent you describe.

'mini.notify' module is meant to show on-demand and LSP notifications. Overriding built-in notifications is out of scope of 'mini.notify', as this is still not an easy to implement and maintain functionality.

That said, you can mimic this behavior by creating an autocommand which gets triggered on every buffer write:

```lua
vim.api.nvim_create_autocmd('BufWritePost', {
  callback = function(args)
    local path = vim.api.nvim_buf_get_name(args.buf)
    if path ~= '' then path = vim.fn.fnamemodify(path, ':~:.') end
    -- Assumes `vim.notify()` overridden by 'mini.notify'
    vim.notify('Saved ' .. vim.inspect(path))
  end,
})
```

**Reply by pwnwriter** [reply]
Thanks ! that works tho, 

Btw, is there any way to disable the message like `the filename` while saving on the bottom?

**Reply by echasnovski** [reply]
I think `vim.opt.shortmess:append('W')` should do that. Or some other 'shortmess' flag.

**Reply by pwnwriter** [reply]
I implemented the following solution to make it work:

As you suggested,

```lua
vim.api.nvim_create_autocmd("BufWritePost", {
  callback = function(args)
    local path = vim.api.nvim_buf_get_name(args.buf)
    if path ~= "" then
      path = vim.fn.fnamemodify(path, ":~:.")
    end
    vim.notify = MiniNotify.make_notify()
    vim.notify("Saved " .. vim.inspect(path))
  end,
})
```

***I set `cmdheight` to `0`***

`vim.opt.cmdheight = 0`

Since cmdheight is set to 0, it doesn't show any info while recording a macro. Therefore, the following autocmd was handy for me:

```lua
function M.autocmds()
  local autocmd = vim.api.nvim_create_autocmd

  autocmd("RecordingEnter", {
    callback = function()
      vim.opt.cmdheight = 1
    end,
  })

  autocmd("RecordingLeave", {
    callback = function()
      vim.opt.cmdheight = 0
    end,
  })
end
```

**Comment by GNOMES** [comment]
is it still not planned/within-scope to show buitin notifications using mini.notify?

**Reply by echasnovski** [reply]
There might be something like this in planned 'mini.cmdline', but that would require more stable command line features in Neovim core.

There is also currently an ongoing work for redoing built-in messages/errors/notifications which might be good enough out of the box. We'll see.
